lib/trino-filesystem/src/main/java/io/trino/filesystem/ParsedLocation.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Character.isWhitespace;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Parse URI {@code scheme://[userInfo@]host[:port][/path]}
+ */
+// This is only present to aid in the development of filesystems, and should not be used by client code
```

Comment:
We need this in Delta Lake's `TransactionLogSynchronizerManager`, for example, to get the scheme